### DISCUSSION TITLE
[Feature Request] Provide alternative for actions/upload-artifact

### DISCUSSION THREAD

**Discussion by hans-d** [runs-on/runs-on#194] [Ideas]
Github artifact storage is "limited" (you can buy more). It would be nice to be able to use self-hosted s3 bucket - like cache .

Eg something like https://github.com/usualdesigner/s3-artifact-upload

**Comment by crohr** [comment]
@hans-d any reason to not use this action directly? You can simply pass `${{ env.RUNS_ON_S3_BUCKET_CACHE }}` as the bucket name, and you might also want to specify a `prefix` input ([supported by the action](https://github.com/usualdesigner/s3-artifact-upload/blob/main/src/main.ts#L52) but not on the README).

Your runners will have the necessary permissions to access the S3 bucket already.

**Comment by hans-d** [comment]
Rather have something from the same source (incl the maintenance on it).

**Comment by crohr** [comment]
I actually had a look at this while developing the [Magic Cache](https://runs-on.com/caching/magic-cache/) feature. It can actually be done, but the issue is that you won't be able to access the artefact links for download directly for the GitHub UI. Which I think breaks the usability. It would be nice if GitHub provided a way to override the download URL when uploading an artefact, but it's not possible yet.

**Comment by crohr** [comment]
Going to close this for now. I don't think it makes sense maintenance-wise to override the upload-artefact mechanism, especially since there is no UI to fetch the resulting artefacts. Probably best to just upload to S3 directly and share the bucket name / download URLs in another way.
